### ISSUE TITLE
Inconsistency in loss calculation between GitHub implementation and Paper Appendix (Temperature scaling)

### ISSUE THREAD

**Issue by Dung-04** [hunto/image_classification_sota#19]
Hi @hunto ,
Thank you for the great work and for releasing the source code. 
While reviewing the implementation of the DIST loss in this repository, I noticed a discrepancy compared to the pseudocode provided in the Appendix of your paper: "Knowledge Distillation from A Stronger Teacher".

Description of the discrepancy
In the GitHub implementation, the inter_loss and intra_loss are scaled by self.tau**2:
#From github implementation 
inter_loss = self.tau **2 * inter_class_relation(y_s, y_t)
intra_loss = self.tau **2 * intra_class_relation(y_s, y_t)

However, in the Paper Appendix section  A.1 Implementation of DIST, these terms are calculated without the $\tau^2$ multiplier:
#From paper appendix
inter_loss = inter_class_relation(y_s, y_t)
intra_loss = intra_class_relation(y_s, y_t)

Questions:
- Is the $\tau^2$ scaling necessary for balancing the gradient magnitude (similar to the original KD loss by Hinton et al.)?
- Which version should be considered the standard reference for reproducing the results mentioned in the paper?
If the $\tau^2$ scaling is intended, I suggest updating the pseudocode in the documentation/paper to avoid confusion for future researchers!
Best regards, Minh Dung Le
